src/Apps/Order/Routes/Review/index.tsx

```diff
@@ -196,6 +196,27 @@ export class ReviewRoute extends Component<ReviewProps, ReviewState> {
           )
           break
         }
+        case "charge_authorization_failed": {
+          let data = {} as any
+          if (error.data) {
+            data = JSON.parse(error.data)
```

Comment:
This was probably my oversight from the `/accept` page, but I'd love to see this wrapped in a try/catch (no need to do anything in the catch block), just in case we get something other than well-formed json back from exchange.